Sources/ProjectDescription/Headers.swift

```diff
@@ -64,4 +70,58 @@ public struct Headers: Codable, Equatable {
             exclusionRule: exclusionRule
         )
     }
+
+    private static func headers(from list: FileList,
+                                umbrella: Path,
+                                private: FileList? = nil,
+                                allOthersAsProject: Bool) -> Headers
+    {
+        .init(
+            public: list,
+            umbrellaHeader: umbrella,
+            private: `private`,
+            project: allOthersAsProject ? list : nil,
+            exclusionRule: .projectExcludesPrivateAndPublic
+        )
+    }
+
+    /// Loading headers from the file list,
+    /// as `public` will be marked all presented in the umbrella header
+    /// as `private` - all from the file list of private headers (exclude already found `public`)
+    /// as `project` - all others
```

Comment:
Trying to clarify a bit:
```suggestion
    /// Headers from the file list are included as:
    /// - `public`, if the header is present in the umbrella header
    /// - `private`, if the header is present in the `private` list
    /// - `project`, otherwise
```